Lib/imaplib.py

```diff
@@ -1279,6 +1389,251 @@ def print_log(self):
                 n -= 1
 
 
+class Idler:
+    """Iterable context manager: start IDLE & produce untagged responses
+
+    An object of this type is returned by the IMAP4.idle() method.
+    It sends the IDLE command when activated by the 'with' statement, produces
+    IMAP untagged responses via the iterator protocol, and sends DONE upon
+    context exit.
+
+    Iteration produces (type, datum) tuples.  They slightly differ
+    from the tuples returned by IMAP4.response():  The second item in the
+    tuple is a single datum, rather than a list of them, because only one
+    untagged response is produced at a time.
```

Comment:
Still, wouldn't it be easier for some user code if the format were the same (just always returning a list of one element)? The fact that you have to warn for this suggests there's something fishy here.

Reply:
The problem with IMAP literal strings is that _imaplib_ divides one response into one or more tuples, terminated by a byte string. Essentially one list item per line plus literal string. Real example of a FETCH response from <https://github.com/python/cpython/issues/68215#issuecomment-2090366295>:

```python
>>> M.fetch('1', '(BODY.PEEK[HEADER.FIELDS (DATE)])')
  56:52.30 > b'NMFB14 FETCH 1 (BODY.PEEK[HEADER.FIELDS (DATE)])'
  56:52.68 < b'* 1 FETCH (BODY[HEADER.FIELDS (DATE)] {41}'
  56:52.68 read literal size 41
  56:52.68 < b')'
  56:52.68 < b'NMFB14 OK Success'
('OK',
 [(b'1 (BODY[HEADER.FIELDS (DATE)] {41}',
   b'Date: Fri, 17 Sep 2004 11:56:09 +1000\r\n\r\n'),
  b')'])
```

See how the one response with embedded literal has been split into data=[(prefix, literal), suffix]. I suspect if a similar IMAP literal was received before or during an IDLE command, your code would yield (prefix, literal) in one iteration, and only yield _suffix_ in the next iteration.